os/pm/pm_procfs.c

```diff
@@ -199,155 +200,77 @@ const struct procfs_operations power_procfsoperations = {
-		if (strcmp(relpath, g_power_direntry[i].name) == 0) {
-			dir->direntry = i;
+	char *str;
+	int domain_id;
+	/* Function to check path is starting with given name */
+	int checkStart(char *name, bool isDirectory) {
+		uint16_t length = strlen(name);
+		if (strncmp(str, name, length) != 0) {
+			return ERROR;
+		}
+		str += length;
+		if (str[0] == '\0') {
 			return OK;
 		}
-	}
-
-	return -ENOENT;
-}
-
-/****************************************************************************
- * Name: power_states_read
- ****************************************************************************/
-
-static size_t power_states_read(FAR struct file *filep, FAR char *buffer, size_t buflen)
-{
-	FAR struct power_file_s *priv;
-	size_t copysize;
-	size_t totalsize;
-	int i;
-
-	priv = (FAR struct power_file_s *)filep->f_priv;
-
-	totalsize = 0;
-
-	if (priv->offset == 0) {
-		/* PM does not support states for specific domain now. It will be updated. */
-		for (i = 0; i < g_power_statescount; i++) {
-			copysize = snprintf(buffer, buflen, "%s ", g_power_states[i]);
-			buflen -= copysize;
-			buffer += copysize;
-			totalsize += copysize;
+		if (isDirectory && (str[0] == '/')) {
+			str++;
+			return OK;
 		}
-
-		/* Indicate we have already provided all the data */
-		priv->offset = 0xFF;
+		return ERROR;
 	}
-
-	return totalsize;
-}
-
-/****************************************************************************
- * Name: power_curstate_read
- ****************************************************************************/
-
-static size_t power_curstate_read(FAR struct file *filep, FAR char *buffer, size_t buflen)
-{
-	FAR struct power_file_s *priv;
-	size_t copysize;
-
-	priv = (FAR struct power_file_s *)filep->f_priv;
-	copysize = 0;
-
-	if (priv->offset == 0) {
-		copysize = snprintf(buffer, buflen, "%s", g_power_states[g_pmglobals.state]);
-		/* Indicate we have already provided all the data */
-		priv->offset = 0xFF;
-	}
-
-	return copysize;
-}
-
-/****************************************************************************
- * Name: power_devices_read
- ****************************************************************************/
-
-static size_t power_devices_read(FAR struct file *filep, FAR char *buffer, size_t buflen)
-{
-	FAR struct power_file_s *priv;
-	FAR struct pm_callback_s *callback;
-	FAR dq_entry_t *entry;
-	size_t copysize;
-	size_t totalsize;
-
-	priv = (FAR struct power_file_s *)filep->f_priv;
-	copysize = 0;
-	totalsize = 0;
-
-	if (priv->offset == 0) {
-
-		entry = dq_peek(&g_pmglobals.registry);
-		while (entry) {
-			callback = (FAR struct pm_callback_s *)entry;
-			copysize = snprintf(buffer, buflen, "%s ", callback->name);
-			buflen -= copysize;
-			buffer += copysize;
-			totalsize += copysize;
-			entry = sq_next(entry);
+	str = relpath;
+	dir->domain_id = -1;
+	/* Check relpath has "power" mount point */
+	if (checkStart(POWER, true) == OK) {
+		if (str[0] == '\0') {
+			dir->base.level = 1;
```

Comment:
What does 1, 2, 3 mean?

Reply:
Here `base.level` represents directory level


```
Level 0 : /proc
Level 1 : /proc/power
Level 2 : /proc/power/domains
Level 3 : /proc/power/domains/{domain_name}
```

Reply:
Let's use some definition for this instead of 1,2,3.

Reply:
done